include/alpaka/queue/QueueHipRtBlocking.hpp

```diff
@@ -348,7 +354,7 @@ namespace alpaka
                 {
                     ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
 
-#if defined( BOOST_COMP_HCC ) && BOOST_COMP_HCC
+#if BOOST_COMP_HCC
```

Comment:
why is `defined( BOOST_COMP_HCC )` not needed anymore?

Reply:
if `BOOST_COMP_HCC` is undefined `#if <undefined>` equals to `#if 0` anyways, so no need to check with defined, as long as we do not do numerical comparisons. Maybe there are compilers which do no t behave like that, maybe you know more?
It is actually defined in the c++ standard (§ 14.1 (11), depends on version):
>  After all replacements due to macro expansion and evaluations of defined-macro-expressions and has-include-expressions have been performed, all remaining identifiers and keywords, except for true and false, are replaced with the pp-number 0